### PR TITLE
[WIP] Initial implementation

### PR DESCRIPTION
**Pull request by Vermyndax**
Almost done, please wait to merge until [WIP] is removed. I want to test this before integrating into the core repo.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Welcome
+
+We're so glad you're thinking about contributing to an 18F open source project! If you're unsure about anything, just ask -- or submit the issue or pull request anyway. The worst that can happen is you'll be politely asked to change something. We love all friendly contributions.
```
Comment (briankfitzwater) [inline CONTRIBUTING.md:3]: This isn't an 18F open source project.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,3 @@
 # devsecops-iam-roles
```
Comment (briankfitzwater) [inline README.md:1]: This could be expanded

**Review by briankfitzwater — COMMENTED** [review]
I feel this is overly complicated and it will be easy for mis-configurations to slip through the cracks.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Vermyndax) [reply CONTRIBUTING.md:3]: Good point. @afeld what do we need to do for the CONTRIBUTING.md file?

File: README.md (hunk 2 above)
Comment (Vermyndax) [reply README.md:1]: Heh.

File: CONTRIBUTING.md (hunk 1 above)
Comment (afeld) [reply CONTRIBUTING.md:3]: I had been using the 18F one out of laziness, but the template to use is actually [the GSA one](https://github.com/GSA/open-source-policy/blob/master/CONTRIBUTING.md).

File: README.md (hunk 2 above)
Comment (afeld) [reply README.md:1]: Especially given @Vermyndax JUST brought up needing to have better user documentation 😛

File: terraform/secops/secops-iam-policies.tf [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+                "route53:*",
+                "ssm:ListAssociations",
+                "ssm:DeleteDocument",
+                "ssm:DescribeAvailablePatches"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
```
Comment (afeld) [inline terraform/secops/secops-iam-policies.tf:144]: Not a blocker, but I recommend saving the policies in the repository as JSON files, and including them using something like

```hcl
policy = "${file("${path.module}/admin_policy.json")}"
```

Makes the JSON syntax-highlighted, and keeps the Terraform shorter / more readable.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+1. [Install the CircleCI CLI.](https://circleci.com/docs/2.0/local-jobs/#installing-the-cli-locally)
+1. Run the tests.
+
+    ```sh
+    make test
+    ```
+
+### Testing in AWS
+
+See [the documentation](docs/integration.md).
```
Comment (afeld) [inline CONTRIBUTING.md:34]: Assuming this is leftover from another repo?

File: terraform/authlanding/authlanding-iam-policies.tf [hunk 5]
```diff
@@ -0,0 +1,589 @@
...
+                "ssm:DeleteAssociation",
+                "ssm:DeregisterManagedInstance",
+                "trustedadvisor:*",
+                "ssm:PutInventory",
+                "sso:*",
+                "lambda:*",
+                "ssm:ListAssociations",
+                "ssm:DeleteDocument",
+                "elasticache:*",
+                "ssm:DescribeAvailablePatches"
```
Comment (afeld) [inline terraform/authlanding/authlanding-iam-policies.tf:135]: Mind sorting the `Actions` alphabetically?

**Review by afeld — DISMISSED** [review]
This pull request is too big to actually comb through in great detail, but assuming you worked on the policies with @horvathy (and maybe others) anyway, I say we merge this now and do smaller follow-up pull requests.

**Review by afeld — APPROVED** [review]
Merging, to do smaller changes on top.

File: terraform/authlanding/authlanding-iam-policies.tf (hunk 5 above)
Comment (afeld) [reply terraform/authlanding/authlanding-iam-policies.tf:135]: Done in #4.

File: terraform/secops/secops-iam-policies.tf (hunk 3 above)
Comment (afeld) [reply terraform/secops/secops-iam-policies.tf:144]: Done in #4.

**Pull request merged**